tree/20_cfe_basics/log_rudder.cf

```diff
@@ -202,3 +224,18 @@ bundle agent clean_reporting_context {
     "report_data.rule_id"        string => "";
     "report_data.identifier"     string => "";
 }
+
+bundle agent report_data {
+  vars:
+    "should_report" string => "true";
```

Comment:
true and false are not classes in cfengine, use any and a class that is garanteed to be never defined

Reply:
or use a cmp() in the class definition in generic methods

Reply:
"true" and "false" can be used to defined classes like done in the other bundle

https://docs.cfengine.com/docs/3.10/reference-promise-types-classes.html#expression

Reply:
your first url point to nothing.
My cfengine test code says it doesn't work:

  reports:
      "Done";
    true::
      "true";
    false::
      "false";

-> Done

Reply:
Here we are defining a variable not a class

Reply:
You should do likewise

Reply:
You should do this way:

vars:

  "should_report" string => "true";

classes:
  "report" expression => "${should_report}";

reports:

  report:: 
    "hey I have reporting based on ${should_report};

Reply:
bundle agent test {

vars:

  "should_report" string => "true";

classes:
  "report" expression => "true";

reports:

  "no  reporting based on ${should_report}";
  report::
  "hey I have reporting based on ${should_report}";
}

Reply:
here we only define the value "should_report"

Reply:
This works since cfengine 3.9

Reply:
All versions of Rudder (4.1 + are using cfengine 3.10)

Reply:
So we have to chose what we want to de here